# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/forms/nsFieldSetFrame.cpp
# Commit: f2c924263ffe
# Full Hash: f2c924263ffece6d0b30e6a52dfd32792c5bf5ab
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2024-01-16 16:37:21
# Regressor Bug: 1765615
# File Overlap Count: 1
# Description:
#   Bug 1765615 - Introduce NS_BLOCK_DYNAMIC_BFC state bit. r=layout-reviewers,emilio
#   
#   After D198523, NS_BLOCK_FLOAT_MGR and NS_BLOCK_MARGIN_ROOT are always
#   set simultaneously when initiating a frame, so we can merge them into a
#   single NS_BLOCK_STATIC_BFC flag. The freed bit is used to define a new
# ==============================================================================

diff -r d72041117125 -r f2c924263ffe layout/forms/nsFieldSetFrame.cpp
--- a/layout/forms/nsFieldSetFrame.cpp	Tue Jan 16 07:00:26 2024 +0000
+++ b/layout/forms/nsFieldSetFrame.cpp	Tue Jan 16 07:58:47 2024 +0000
@@ -788,7 +788,7 @@
   if (nsBlockFrame* legend = do_QueryFrame(GetLegend())) {
     // A rendered legend always establish a new formatting context.
     // https://html.spec.whatwg.org/multipage/rendering.html#rendered-legend
-    legend->AddStateBits(NS_BLOCK_FORMATTING_CONTEXT_STATE_BITS);
+    legend->AddStateBits(NS_BLOCK_STATIC_BFC);
   }
   MOZ_ASSERT(
       aListID != FrameChildListID::Principal || GetInner() || GetLegend(),
@@ -819,7 +819,7 @@
   if (nsBlockFrame* legend = do_QueryFrame(GetLegend())) {
     // A rendered legend always establish a new formatting context.
     // https://html.spec.whatwg.org/multipage/rendering.html#rendered-legend
-    legend->AddStateBits(NS_BLOCK_FORMATTING_CONTEXT_STATE_BITS);
+    legend->AddStateBits(NS_BLOCK_STATIC_BFC);
   }
 }
 